common/src/main/java/org/conscrypt/OpenSSLX509CertificateFactory.java

```diff
@@ -60,6 +60,43 @@
         }
     }
 
+    private static boolean isMaybePkcs7(byte[] header) {
+        // The outer tag must be SEQUENCE.
+        if (header.length < 2 || header[0] != 0x30) {
+            return false;
+        }
+
+        // Bytes are signed in Java.
+        int lengthByte = header[1];
```

Comment:
Signed bytes are indeed ugly, but the more usual Java idiom here is `int lengthByte = header[1] & 0xff;` rather than an extra conditional.

Reply:
Ah, nice. Done.